### ISSUE TITLE
Google Tag Manager CORS error

### ISSUE THREAD

**Issue by gbvaz** [ahmedkaludi/accelerated-mobile-pages#4528]
Hi, I’m using AMP for WP plugin, and I’m getting this error since the last update, version 1.0.52, I guess I skipped some versions since the last time I’ve updated, my last production version was 0.9.98.2.

URL for testing:
https://educandoseubolso.blog.br/2020/06/01/maquina-de-cartao-mercado-pago/amp/

It happens no just for this one, but for all AMP pages in my website.

Console log:
`Access to fetch at 'https://www.googletagmanager.com/amp.json?id=%20GTM-NHSLHRS&__amp_source_origin=https%3A%2F%2Feducandoseubolso.blog.br' from origin 'https://educandoseubolso.blog.br' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

I can assure I’m using a Google Tag Manager container for AMP, following Google’s guidelines for this matter. Since the plugin update, Google Analytics is no longer showing pageviews for AMP pages, I’m assuming it’s because of this error.

Screenshots from my Tag Manager account settings:

https://imgur.com/a/clpj4pd

I've submitted a ticket in WP forum as well (https://wordpress.org/support/topic/google-tag-manager-cors-error/) so no worries, please continue with the investigation wherever you guys feel more confortable. How can I help?

**Comment by gbvaz** [comment]
I'm not sure, but according the error message, there's no "Access-Control-Allow-Origin" header. I did a quick search in the plugin code, and only found one mention, inside template-mode.php file, in a function related to ajax comments. Coincidentally I had disabled comments for posts in my amp settings, so my guess is that it is only attaching this header once this option is enabled, but I'm not sure. I've enabled it back again, cleared my website's cache and I'm at this moment regenerating all cache files with WP-Rocket plugin. As far as I can see, the problem still persists, although I'm noticing some amp pages showing up in Google Analytics real-time report, and that's something that wasn't happening at all in last few days.

**Comment by ahmedkaludi** [comment]
@gbvaz can you reach out to us via our website?

We need to verify this bug first.

**Comment by gbvaz** [comment]
Just a quick update, we've exchanged a few emails, and everything looks ok in my WP and Tag Manager configuration. We are using AWS EC2 instances for this WP install under Apache, and I've reached out to AWS support with this problem. I'm not sure they are going to provide any help for this matter, but it's worth the shot. My guess is this could be something related to the plugin itself, because everything was working fine since the last time I've updated it, but it could also be something related to Google services, since usually CORS configuration are made in the host, in this case Tag Manager.
